src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+
+  let currentState = { ...state };
+
+  for (const action of actions) {
+
+    if (action.type === 'clear') {
+      currentState = {};
+
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+
+    } else if (action.type === 'removeProperties') {
+
+      const keysToRemove = action.keysToRemove || [];
+      for (const key of keysToRemove) {
+        delete currentState[key];
+      }
+    }
```

Comment:
`else / if` works here, but `switch / case` would be way simpler and readable solution
Please rewrite it with a switch